meshio/msh_io/main.py

```diff
@@ -22,22 +22,41 @@ def read_buffer(f):
     int_size = sizeof(c_int)
     format_version, data_size, is_ascii = _read_header(f, int_size)
 
-    assert format_version in ["2", "4"], "Need mesh format 2 or 4 (got {})".format(
-        format_version
-    )
+    readers = {2: msh2, 4.0: msh4_0, 4.1: msh4_1}
 
-    reader = {"2": msh2, "4": msh4}[format_version]
+    try:
+        reader = readers[format_version]
+    except KeyError:
+        try:
+            reader = readers[int(format_version)]
+        except KeyError:
+            raise ValueError(
+                "Need mesh format in {} (got {})".format(readers.keys(), format_version)
+            )
 
     return reader.read_buffer(f, is_ascii, int_size, data_size)
 
 
 def _read_header(f, int_size):
+    """Read the mesh format block
+
+    specified as
+
+     version(ASCII double; currently 4.1)
```

Comment:
See above. -> str

Reply:
Yes.  That's just quoting the spec for [MSH file format (version 4)](http://gmsh.info/dev/doc/texinfo/gmsh.html#MSH-file-format-_0028version-4_0029), but I added a line noting that here it's kept as str. e32ccf795d630ef340b106dd91f0be31d8d045bb

Reply:
> version(ASCII double; currently 4.1)

Smh. We'll stick with strings.